app/controllers/clients/vispdats_controller.rb

```diff
@@ -5,6 +5,7 @@ class VispdatsController < Window::Clients::VispdatsController
     before_action :require_can_edit_vspdat!
 
     def destroy
+      @vispdat.disassociate_files
```

Comment:
Any reason not to put this in a before_destroy?

Reply:
I initially tried that, but something about when the foreign key association is checked prevented it from working.